### ISSUE TITLE
Add support for Single Select Combobox

### ISSUE THREAD

**Issue by koolamusic** [koolamusic/chakra-ui-autocomplete#2]
This will be a wrapper around https://downshift.netlify.app/use-combobox
Will document thoughts while at it

**Comment by it-nalon** [comment]
I'm trying to figure out how this can be implemented.

For now, I found two ways to achieve the goal:

### Incrementally adopted implementation
A first minor release that adds the `multiple: boolean` prop to the `CUIAutocomplete` component.
A second major release that just ships two different components that you can import like this: `import {CUIAutocomplete, CUIMultipleAutocomplete} from 'chakra-ui-autocomplete'`.

I think that we can implement the minor release splitting the actual `index.tsx` in three different files:
- **CUIAutocomplete.tsx/Autocomplete.tsx**: single select combobox  implemented with `useCombobox()` hook
- **CUIMultipleAutocomplete.tsx/MultipleAutocomplete.tsx**: current implementation of the Autocomplete
- **index.tsx**: a public `Autocomplete` component that add the `multiple` prop used to switch between the two components.

**Pro**: no breaking changes.
**Cons**: complicate typings in minor release.


### Major release only
Like the "_incrementally adopted implementation_" but without the "_minor release step_", so just a new major version that ships two components instead of the actual one.

**Pro**: Simpler to implement.
**Cons**: complete breaking changes.

What do you think?

**Comment by koolamusic** [comment]
I'll go for a breaking change, as long as we can provide comprehensive documentation, the other versions of the app provide the compatibility required, because from here on moving forward, this will create the opportunity add features that are relevant to both workflows

**Comment by it-nalon** [comment]
Hi everyone, I started working on the `CUIAutoComplete`+`CUIMultipleAutoComplete` components.
Right now I have a working example that I'm already using in one application.

It's still a big WIP full of copied code and other ugly things, I'll try to fix everything in the next few days.

The branch is here: [feat-single-select-combobox](/Hyperting/chakra-ui-autocomplete/tree/feat-single-select-combobox)

**Comment by koolamusic** [comment]
Thanks @it-nalon for taking this up, much appreciated. I reckon when you do get the chance to send a PR we can take a look at it.

**Comment by it-nalon** [comment]
Hi everyone, I'm not disappeared.
We have done some changes to the CUIAutoComplete components and we've tested everything all the week.
We need some time to document the new props and add some use cases in the README.

I have also some suggestions to propose: upgrade tsdx to the latest version, add Storybook, change the name: "AutoComplete" to "Autocomplete", etc.

Maybe we can make a new branch with a npm version (a "next" version/Github label?) where we can work togheter on this new version of the library?
@koolamusic what do you think?

**Comment by koolamusic** [comment]
Yeah, you beat me to storybook 😆 . let's create the `next` branch and work from there. these changes are welcome. Thanks for all the hard work on this. Much appreciated 🤝

**Comment by koolamusic** [comment]
@it-nalon any updates on this. let me know if you need some assistance

**Comment by it-nalon** [comment]
> @it-nalon any updates on this. let me know if you need some assistance

Hi @koolamusic, sorry for the delay.
I'll restart working on this project this weekend.
You can check the progress on my branch (or we can merge it here in this repository).

I correctly added Storybook and it works like a charm, just missing the documentation, the tests and the "Example" project in which I found some difficulties to make it run correctly.

We're already using this version in production since 1 month and it works perfectly for now.

Sorry again for the delay!

**Comment by koolamusic** [comment]
sure, sounds good to me. 
What I think we might do is eliminate the examples project in favour of storybook and use that as demo for now until we figure out documentation. I'd prefer codesandbox sample projects and some decent docs atm. 

Storybook is important right now as I wrote this for a project I was working on, however to start adding more features in isolation, I think it would be extremely beneficial to adopt storybook from now on-wards.

Meanwhile thanks for all the hard work on this one. Much appreciated

**Comment by it-nalon** [comment]
Great!
Should I make a PR on `dev` branch o similiar?
I can make it before the weekend with some instructions in the `README.md` to run everything, but as I have already said without documentation and tests.

**Comment by koolamusic** [comment]
sure. feel free
